src/Cookie.php

```diff
@@ -4,148 +4,150 @@
+		$this->domain = $domain;
+		$this->secure = $secure;
+		$this->httpOnly = $httpOnly;
+		$this->options = $options;
+	}
 
-    public function getName(): string {
-        return $this->name;
-    }
+	public function getName(): string {
+		return $this->name;
+	}
 
-    public function setName(string $name): void {
-        if ('' === $name) {
-            return;
-        }
+	public function setName(string $name): void {
+		if ('' === $name) {
+			return;
+		}
 
 		$this->name = $name;
-    }
+	}
 
-    public function getValue(): string {
-        return $this->value;
-    }
+	public function getValue(): string {
+		return $this->value;
+	}
 
-    public function setValue(string $value): void {
-        $this->value = $value;
-    }
+	public function setValue(string $value): void {
+		$this->value = $value;
+	}
 
-    public function getExpires(): int {
-        return $this->expires;
-    }
+	public function getExpires(): int {
+		return $this->expires;
+	}
 
-    public function setExpires(int $expires): void {
-        $this->expires = $expires;
-    }
+	public function setExpires(int $expires): void {
+		$this->expires = $expires;
+	}
 
-    public function getPath(): string {
-        return $this->path;
-    }
+	public function isExpired(): bool {
+		return $this->expires < time();
+	}
 
-    public function setPath(string $path): void {
-        $this->path = $path;
-    }
+	public function getRemainingTime(): int {
+		return $this->expires - \time();
+	}
 
-    public function getDomain(): string {
-        return $this->domain;
-    }
+	public function getPath(): string {
+		return $this->path;
+	}
 
-    public function setDomain(string $domain): void {
-        $this->domain = $domain;
-    }
+	public function setPath(string $path): void {
+		$this->path = $path;
+	}
 
-    public function getSecure(): bool {
-        return $this->secure;
-    }
+	public function getDomain(): string {
+		return $this->domain;
+	}
 
-    public function setSecure(bool $secure): void {
-        $this->secure = $secure;
-    }
+	public function setDomain(string $domain): void {
+		$this->domain = $domain;
+	}
 
-    public function getHttpOnly(): bool {
-        return $this->httpOnly;
-    }
+	public function isSecure(): bool {
+		return $this->secure;
+	}
+
+	public function setSecure(bool $secure): void {
+		$this->secure = $secure;
+	}
+
+	public function getHttpOnly(): bool {
+		return $this->httpOnly;
+	}
 
-    public function setHttpOnly(bool $httpOnly): void {
-        $this->httpOnly = $httpOnly;
-    }
+	public function setHttpOnly(bool $httpOnly): void {
+		$this->httpOnly = $httpOnly;
+	}
 
-    public function getOptions(): array {
-        return $this->options;
-    }
+	public function getOptions(): array {
+		return $this->options;
+	}
 
-    public function setOptions(array $options): void {
-        $this->options = $options;
-    }
+	public function setOptions(array $options): void {
+		$this->options = $options;
```

Comment:
Should validate the array (See possible options at https://www.php.net/manual/de/function.setcookie.php)